src/server/service/leaderboard/worker-client.ts

```diff
@@ -0,0 +1,24 @@
+import path from "node:path";
+import { Worker } from "node:worker_threads";
+import { fileURLToPath } from "node:url";
+
+import { type UpdateScoreArgs } from "./types.js";
+
+const updateScores = (args: UpdateScoreArgs) => {
+  return new Promise((resolve, reject) => {
+    // `__dirname` is not available when using ESM in Node
+    const filename = fileURLToPath(import.meta.url);
+    const workerPath = path.join(filename, "..", "worker.js");
```

Comment:
It may be worth to add a comment in the worker.js/ts file saying that there exist a hardcoded reference to the name of that file somewhere else...